client/components/app.jsx

```diff
@@ -9,24 +9,34 @@ class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      restaurantId: 1001,
+      restaurantId: 1001, 
       timeSlots: [],
       partySize: 2,
+      partySizeMax: 15,
       date: "2018-05-31",
-      time: "5:00 am",
+      time: "5:00 PM",
       pickTime: "",
       clicked: false,
     }
-  } 
+  }   
+  componentDidMount() {
+    var context = this;
+    axios.get('/restaurant/1001/' + this.state.date)
+      .then(function(response) {
+        context.setState({
+          timeSlots: response.data,
+        });
+      });
+  }
   findTable () {
     var context = this; 
```

Comment:
Refactoring to ES6 function would allow you to get rid of this variable. Due to the callback taking the this binding of the parent fxn.